File: test/python/service_client/app/src/launcher.py
# Copyright (c) 2024-2025 Contributors to the Eclipse Foundation
#
# This program and the accompanying materials are made available under the
# terms of the Apache License, Version 2.0 which is available at
# https://www.apache.org/licenses/LICENSE-2.0.
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
# License for the specific language governing permissions and limitations
# under the License.
#
# SPDX-License-Identifier: Apache-2.0


from vcsptcpbylimservice_service_sdk import VCSPtCpbyLimServiceStub
from vcsptcpbylimservice_service_sdk import VCSPtCpbyLimServiceServiceClientFactory
from vcsptcpbylimservice_service_sdk.motorcontrol_pb2 import NtfPtPwrLimRequest
from vcsmotortrqmngservice_service_sdk import VCSMotorTrqMngServiceStub
from vcsmotortrqmngservice_service_sdk import VCSMotorTrqMngServiceServiceClientFactory
from vcsmotortrqmngservice_service_sdk.motorcontrol_pb2 import SetMCUCtrlReqRequest
from val_service_sdk.val_pb2_grpc import VALStub
from val_service_sdk import VALServiceClientFactory
from val_service_sdk.val_pb2 import GetRequest
from hornservice_service_sdk.horn_pb2 import IsRunningRequest
from hornservice_service_sdk.horn_pb2_grpc import HornServiceStub
from hornservice_service_sdk import HornServiceServiceClientFactory
from seats_service_sdk.seats_pb2_grpc import SeatsStub
from seats_service_sdk.seats_pb2 import MoveRequest
from seats_service_sdk.SeatsServiceClientFactory import SeatsServiceClientFactory
from velocitas_sdk.base import Middleware
from velocitas_sdk import config


def create_seats_client(middleware: Middleware) -> SeatsStub:
    client = SeatsServiceClientFactory.create(middleware)

    return client


def create_horn_client(middleware: Middleware) -> HornServiceStub:
    client = HornServiceServiceClientFactory.create(middleware)

    return client


def create_motorcontrol_client(middleware: Middleware) -> VCSMotorTrqMngServiceStub:
    client = VCSMotorTrqMngServiceServiceClientFactory.create(middleware)

    return client


def create_capacitycontrol_client(middleware: Middleware) -> VCSPtCpbyLimServiceStub:
    client = VCSPtCpbyLimServiceServiceClientFactory.create(middleware)

    return client


def create_val_client(middleware: Middleware) -> VALStub:
    client = VALServiceClientFactory.create(middleware)

    return client


if __name__ == "__main__":
    client_seats = create_seats_client(config.middleware)
    move_request = MoveRequest()
    client_seats.Move(move_request)

    client_horn = create_horn_client(config.middleware)
    is_running_request = IsRunningRequest()
    client_horn.IsRunning(is_running_request)

    client_motorcontrol = create_motorcontrol_client(config.middleware)
    motorcontrol_request = SetMCUCtrlReqRequest()
    client_motorcontrol.SetMCUCtrlReq(motorcontrol_request)

    client_capacitycontrol = create_capacitycontrol_client(config.middleware)
    capacitycontrol_request = NtfPtPwrLimRequest()
    client_capacitycontrol.NtfPtPwrLim(capacitycontrol_request)

    client_val = create_val_client(config.middleware)
    get_request = GetRequest()
    client_val.Get(get_request)

File: test/cpp/service_client/app/src/Launcher.cpp
/**
 * Copyright (c) 2023-2024 Contributors to the Eclipse Foundation
 *
 * This program and the accompanying materials are made available under the
 * terms of the Apache License, Version 2.0 which is available at
 * https://www.apache.org/licenses/LICENSE-2.0.
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

#include <sdk/middleware/Middleware.h>
#include <services/hornservice/HornserviceServiceClientFactory.h>
#include <services/hornservice/horn.grpc.pb.h>
#include <services/seats/SeatsServiceClientFactory.h>
#include <services/seats/seats.grpc.pb.h>
#include <services/val/kuksa/val/v1/ValServiceClientFactory.h>
#include <services/val/kuksa/val/v1/val.grpc.pb.h>
#include <services/vcsmotortrqmngservice/VcsmotortrqmngserviceServiceClientFactory.h>
#include <services/vcsmotortrqmngservice/motorcontrol.grpc.pb.h>
#include <services/vcsptcpbylimservice/VcsptcpbylimserviceServiceClientFactory.h>
#include <services/vcsptcpbylimservice/capacitylimit.grpc.pb.h>

#include <iostream>

#include "SampleApp.h"

using namespace velocitas;

int main(int argc, char** argv) {
   auto seatService = SeatsServiceClientFactory::create(Middleware::getInstance());

   ::grpc::ClientContext seat_context;
   ::sdv::edge::comfort::seats::v1::MoveRequest move_request;
   ::sdv::edge::comfort::seats::v1::MoveReply move_response;
   auto status = seatService->Move(&seat_context, move_request, &move_response);

   std::cout << "gRPC Server returned code: " << status.error_code() << std::endl;
   std::cout << "gRPC error message: " << status.error_message().c_str() << std::endl;

   if (status.error_code() != ::grpc::StatusCode::UNIMPLEMENTED)
   {
      return 1;
   }

   auto hornService =
       HornserviceServiceClientFactory::create(Middleware::getInstance());

   ::grpc::ClientContext hornContext;
   ::bcm::horn::v1::StartRequest start_request;
   ::bcm::horn::v1::StartResponse start_response;

   status = hornService->Start(&hornContext, start_request, &start_response);

   std::cout << "gRPC Server returned code: " << status.error_code() << std::endl;
   std::cout << "gRPC error message: " << status.error_message().c_str() << std::endl;

   if (status.error_code() != ::grpc::StatusCode::UNIMPLEMENTED)
   {
      return 1;
   }

   auto valService = ValServiceClientFactory::create(Middleware::getInstance());

   ::grpc::ClientContext valContext;
   ::kuksa::val::v1::GetRequest get_reqeuest;
   ::kuksa::val::v1::GetResponse get_response;

   status = valService->Get(&valContext, get_reqeuest, &get_response);

   std::cout << "gRPC Server returned code: " << status.error_code() << std::endl;
   std::cout << "gRPC error message: " << status.error_message().c_str() << std::endl;

   if (status.error_code() != ::grpc::StatusCode::UNIMPLEMENTED)
   {
      return 1;
   }

   auto motorcontrolService = VcsmotortrqmngserviceServiceClientFactory::create(
       Middleware::getInstance());

   ::grpc::ClientContext motorcontrolContext;
   ::vcs::powertrain::v1::SetMCUCtrlReqRequest set_mcu_request;
   ::vcs::powertrain::v1::SetMCUCtrlReqResponse set_mcu_response;

   status = motorcontrolService->SetMCUCtrlReq(&motorcontrolContext, set_mcu_request,
                                               &set_mcu_response);

   std::cout << "gRPC Server returned code: " << status.error_code() << std::endl;
   std::cout << "gRPC error message: " << status.error_message().c_str() << std::endl;

   if (status.error_code() != ::grpc::StatusCode::UNIMPLEMENTED)
   {
      return 1;
   }

   auto capacityService = VcsptcpbylimserviceServiceClientFactory::create(
       Middleware::getInstance());

   ::grpc::ClientContext capacitycontrolContext;
   ::vcs::powertrain::v1::NtfPtPwrLimRequest pwr_lim_request;
   ::vcs::powertrain::v1::NtfPtPwrLimResponse pwr_lim_response;

   status = capacityService->NtfPtPwrLim(&capacitycontrolContext, pwr_lim_request,
                                         &pwr_lim_response);

   std::cout << "gRPC Server returned code: " << status.error_code() << std::endl;
   std::cout << "gRPC error message: " << status.error_message().c_str() << std::endl;

   if (status.error_code() != ::grpc::StatusCode::UNIMPLEMENTED)
   {
      return 1;
   }

   return 0;
}
